src/DatabaseRuleSet.php

```diff
@@ -41,6 +42,30 @@ public function exists(string $table, string $column = 'NULL', \Closure $callbac
         return $this->appendIfNotExists((string)$rule);
     }
 
+    /**
+     * Get a exists constraint builder instance by model class.
+     *
+     * @param string $modelClass Enumeration to validate against
+     * @param \Closure|null $callback callback, that will receive \Illuminate\Validation\Rules\Exists $rule
+     * @return GenericRuleSet
+     * @see \Illuminate\Validation\Rules\Exists
+     * @see DatabaseRuleSet::exists
+     */
+    public function modelExists(string $modelClass, \Closure $callback = null): GenericRuleSet
+    {
+        if (!is_a($modelClass, Model::class, true)) {
+            throw new \UnexpectedValueException('Class is not a Model subclass');
```

Comment:
Just a note: IMHO it makes sense to a TypeMismatchException to saritasa/php-common for such cases.